src/swarm-tui/Swarm/TUI/View/Robot.hs

```diff
@@ -28,17 +39,159 @@ import Swarm.Game.State.Substate
+    , _libRenderers =
+        MixedRenderers
+          { cell = dc uiDebug
+          , rowHdr = Just rowHdr
+          , colHdr = Just $ colHdr uiDebug
+          , colHdrRowHdr = Just $ ColHdrRowHdr $ \_ _ -> vLimit 1 (fill ' ') <=> hBorder
+          }
+    }
+
+renderTheRobots :: RobotListContent -> Widget Name
+renderTheRobots rd =
+  renderMixedTabularList (rd ^. libRenderers) (LstFcs True) (rd ^. libList)
+
+columnHdrAttr :: AttrName
+columnHdrAttr = attrName "columnHeader"
+
+rowHdrAttr :: AttrName
+rowHdrAttr = attrName "rowHeader"
+
+colHdr :: Set DebugOption -> MixedColHdr Name Widths
+colHdr uiDebug =
+  MixedColHdr
+    { draw = \_ (MColC (Ix ci)) -> case hdrs V.!? ci of
+        Just ch -> withAttr columnHdrAttr (padRight Max $ str ch) <=> hBorder
+        Nothing -> emptyWidget
+    , widths = \(Widths ws) -> zipWith (<>) ws (map (ColW . length) $ V.toList hdrs)
+    , height = ColHdrH 2
+    }
+ where
+  hdrs = colHdrs uiDebug
+
+-- | Enumerates the rows by position (not 'RID').
+rowHdr :: RowHdr Name RobotWidgetRow
+rowHdr =
+  RowHdr
+    { draw = \_ (WdthD wd) (RowHdrCtxt (Sel s)) rh ->
+        let attrFn =
+              if s
+                then id
+                else withAttr rowHdrAttr
+         in attrFn $ padRight (Pad $ if wd > 0 then 0 else 1) $ padLeft Max (str $ show rh)
+    , width = \_ rh -> RowHdrW . (+ 2) . maximum0 $ map (length . show) rh
+    , toRH = \_ (Ix i) -> i + 1
+    }
+
+headingStringList :: [String]
+headingStringList =
+  map ($ headingStrings) accessorList
+ where
+  headingStrings =
+    LibRobotRow
+      { _fID = "ID"
+      , _fName = "Name"
+      , _fAge = "Age"
+      , _fPos = "Pos"
+      , _fItems = "Items"
+      , _fStatus = "Status"
+      , _fActns = "Actns"
+      , _fCmds = "Cmds"
+      , _fCycles = "Cycles"
+      , _fActivity = "Activity"
+      , _fLog = "Log"
+      }
+
+dropFirstColumn :: Set DebugOption -> [a] -> [a]
+dropFirstColumn uiDebug =
+  applyWhen (not debugRID) (drop 1)
+ where
+  debugRID = uiDebug ^. Lens.contains ListRobotIDs
+
+colHdrs :: Set DebugOption -> Vector String
+colHdrs uiDebug = V.fromList $ dropFirstColumn uiDebug headingStringList
+
+accessorList :: [LibRobotRow a -> a]
+accessorList =
+  [ _fID
+  , _fName
+  , _fAge
+  , _fPos
+  , _fItems
+  , _fStatus
+  , _fActns
+  , _fCmds
+  , _fCycles
+  , _fActivity
+  , _fLog
+  ]
+
+dc :: Set DebugOption -> ListFocused -> MixedCtxt -> RobotWidgetRow -> Widget Name
+dc uiDebug _ (MxdCtxt _ (MColC (Ix ci))) r =
+  maybe emptyWidget (renderPlainCell . wWidget . ($ rPayload r)) (indexedAccessors V.!? ci)
+ where
+  indexedAccessors = V.fromList accessors
+  accessors = dropFirstColumn uiDebug accessorList
+  renderPlainCell = padRight Max
+
+-- | For a single-constructor datatype like 'RobotWidgetRow',
+-- this implementation is trivial.
+wpr :: WidthsPerRow RobotWidgetRow Widths
+wpr = WsPerR $ \(Widths x) _ -> x
+
+wprk :: Set DebugOption -> WidthsPerRowKind RobotWidgetRow Widths
```

Comment:
Explain how this works: we take the maximum of all cell widths, **including the headers**, and add `1` for "padding".